File: simple-py/bfs/bfs.py
from collections import deque
from typing import List


def bfsOfGraph(V: int, adj: List[List[int]]) -> List[int]:

    bfs: List[int] = []
    queue = deque()
    visited: List[bool] = [False] * V

    queue.append(0)

    while queue:
        node = queue.popleft()
        bfs.append(node)

        for adjNode in adj[node]:
            if not visited[adjNode]:
                visited[adjNode] = True
                queue.append(adjNode)

    return bfs


print(bfsOfGraph(5, [[1, 2, 3], [], [4], [], []]))

File: simple-java/src/graphs/BFS.java
package graphs;

import java.util.ArrayList;
import java.util.LinkedList;
import java.util.Queue;

public class BFS {
    public static void main(String[] args) {

    }

    public ArrayList<Integer> bfsOfGraph(int V, ArrayList<ArrayList<Integer>> adj) {
        ArrayList<Integer> bfs = new ArrayList<>();
        boolean[] visited = new boolean[V];
        Queue<Integer> queue = new LinkedList<Integer>();

        queue.add(0);

        while (!queue.isEmpty()) {
            var node = queue.poll();
            bfs.add(node);

            for (Integer it : adj.get(node)) {
                if (visited[it] == false) {
                    visited[it] = true;
                    queue.add(it);
                }
            }
        }

        return bfs;
    }
}
